core/src/main/scala/net/ruippeixotog/scalascraper/browser/JsoupBrowser.scala

```diff
@@ -30,7 +31,11 @@ import net.ruippeixotog.scalascraper.util._
   * @param proxy
   *   an optional proxy configuration to use
   */
-class JsoupBrowser(val userAgent: String = "jsoup/1.8", val proxy: JavaProxy = null) extends Browser {
+class JsoupBrowser(
+    val userAgent: String = "jsoup/1.8",
+    val proxy: JavaProxy = null,
+    val parser: Parser = Parser.htmlParser()
```

Comment:
The reason why your code isn't working is because jsoup's `Parser` is a stateful object. You are not supposed to be reusing the same parser in multiple documents, which is why you're getting all those validation errors and null pointers.

You need to change this to a builder function and create a new instance every time you parse a new document. 

```suggestion
    val parserBuilder: () => Parser = () => Parser.htmlParser()
```

Reply:
Amazing, this is not intuitive in first impressions. If you don't tell me, I wouldn't have dreamed of this place. Nice catch!